common/src/main/scala/explore/common/UserPreferencesQueries.scala

```diff
@@ -100,19 +123,105 @@ object UserPreferencesQueries {
       } yield w).value.map(_.flatten.getOrElse(defaultValue))
   }
 
-  def storeWidthPreference[F[_]: Effect](
-    userId:  Option[User.Id],
-    section: ResizableSection,
-    width:   Int
-  )(implicit
-    cl:      GraphQLClient[F, UserPreferencesDB]
-  ): Callback =
-    userId.map { i =>
-      UserWidthsCreation
-        .execute[F](
-          WidthUpsertInput(i, section, width)
-        )
-        .attempt
-        .runAsyncAndForgetCB
-    }.getOrEmpty
+  /**
+   * Read the grid layout for a given section
+   */
+  @GraphQL(debug = false)
+  object UserGridLayoutQuery extends GraphQLOperation[UserPreferencesDB] {
+    val document = """
+      query layout_positions($criteria: grid_layout_positions_bool_exp!) {
+        grid_layout_positions(where: $criteria) {
+          breakpoint_name
+          height
+          width
+          x
+          y
+          tile
+        }
+      }"""
+
+    def positions2LayoutMap(
+      g: (BreakpointName, List[Data.GridLayoutPositions])
+    ): (react.gridlayout.BreakpointName, (JsNumber, JsNumber, Layout)) =
+      breakpointNameFromString(g._1) -> ((1,
+                                          1,
+                                          Layout(
+                                            g._2.map(p =>
+                                              LayoutItem(p.width, p.height, p.x, p.y, p.tile)
+                                            )
+                                          )
+                                         )
+      )
+
+    // Gets the layout of a section.
+    // This is coded to return a default in case
+    // there is no data or errors
+    def queryWithDefault[F[_]: MonadError[*[_], Throwable]](
+      userId:       Option[User.Id],
+      area:         GridLayoutSection,
+      defaultValue: LayoutsMap
+    )(implicit cl:  GraphQLClient[F, UserPreferencesDB]): F[LayoutsMap] =
+      (for {
```

Comment:
I'm not sure the monad transformer is really needed here.

Isn't the same achieved with `userId.fold(F.pure(defaultValue){uid => ...}` ?

Reply:
Not completely, I want that errors also are handled and go to the default value